tests/bluetooth/tester/src/btp_gatt.c

```diff
@@ -2290,6 +2290,26 @@ static uint8_t get_attr_val_rp(const struct bt_gatt_attr *attr, uint16_t handle,
 		net_buf_simple_add(buf, read);
 	} while (read == to_read);
 
+	/* use userdata only for tester own attributes */
+	if (attr >= &server_db[0] && attr <= &server_db[SERVER_MAX_ATTRIBUTES - 1]) {
+		const struct gatt_value *value = attr->user_data;
+
+		if ((rp->att_response == 0) && (value->enc_key_size > 0)) {
+			/* if keysize is set it means it is used for testing, may be called
+			 * before connection or pairing so just assume this will correctly
+			 * reported for verification. Note that we return "expected"
+			 * error code and data as this is used by autopts validation.
+			 */
```

Comment:
Not sure if it is too early in the morning, or if this comment is confusing to read, but either way I did not understand the comment. 

In what scenario can the connection be NULL where we want to return `BT_ATT_ERR_ENCRYPTION_KEY_SIZE`?

Does `value->enc_key_size` indicate the minimum key size, or the exact key size expected? If the latter, `!=` should be used instead of `>`

Reply:
That command is used to read local test database by autopts, so we need to know which characteristic is suppose to be used by key size tests (we need to provide handle to PTS).

It is min key size and is used to trigger error on GATT operation when PTS pairs with small key size (typically we set it to 16 for specified test characteristics, while PTS pairs with keysize set to <16)

And depending on test, PTS may ask about handle during connection or prior to connection

Reply:
Thanks - That makes sense :) 

Please have a re-read at the in-code comment as well, and see if it needs any updates. Your comment here is much more clear to me at least